src/analyses/interval_domain.h

```diff
@@ -40,10 +40,15 @@ class interval_domaint:public ai_domain_baset
     const ai_baset &ai,
     const namespacet &ns) const final;
 
+  bool join(const interval_domaint &b);
```

Comment:
Merge's signature is fixed and requires location arguments.  It seemed better to split merge into two (the common interface and join, what is actually does the update) and use join when I wanted to compute the join of two abstract domains, independently of having their location, rather than make up false locations.

Using join (outside of merge) is to help compute ai_simplify in the conjunctive case.